data/avro-schema/src/main/resources/avro/net/corda/data/flow/state/checkpoint/SavedOutputs.avsc

```diff
@@ -0,0 +1,21 @@
+{
+  "type": "record",
+  "name": "SavedOutputs",
+  "namespace": "net.corda.data.flow.state.checkpoint",
+  "doc": "Record the output events for a given input event, to be replayed by the flow engine if necessary.",
+  "fields": [
+    {
+      "name": "inputEventHash",
+      "type": "string",
```

Comment:
Is there a reason not to use bytes?

Reply:
It will be passed to the flow engine as a string from the mediator

Reply:
Maybe it's better not to base it on current implementation of  mediator that works with String only because it's stored with Record that allows only String headers. It might be better to represent it with bytes (it will also take less space).

Also, we shouldn't base hashes on UUIDs as it's a bit confusing and we might get in trouble if its implementation changes in the future. Instead we should use a proper hashing algorithm e.g. MD5.

Reply:
For the purposes of this pr, i think it will have to remain a string. as changing the mediator message processor to accept things other than Record<K, V> is not in scope.

I will refactor the runtime to remove UUIDs